### PR TITLE
Fix p5-Text-Xslate on Perl 5.8.5

### LINKED ISSUE #140: 3.54 breaks p5-Text-Xslate
https://github.com/xslate/p5-Text-Xslate/issues/198

@pali can you look at this?

### PR DESCRIPTION
**Pull request by atoomic**
Fixes #140

This is a patch from @pali, to fix the
failure seen by @skaji on p5-Text-Xslate.

### CODE REVIEW

**Comment by atoomic** [comment]
@khwilliamson let's wait on a clean smoker from Travis CI before merging it

File: parts/inc/Sv_set [hunk 1]
```diff
@@ -25,15 +25,15 @@ sv_setsv_flags
...
-    if (((flags) & SV_NOSTEAL) && (SvFLAGS((sstr)) & SVs_TEMP)) {        \
-      SvTEMP_off((sstr));                                                \
-      Perl_sv_setsv_flags(aTHX_ (dstr), (sstr), (flags) & ~SV_NOSTEAL);  \
-      SvTEMP_on((sstr));                                                 \
-    } else {                                                             \
-      Perl_sv_setsv_flags(aTHX_ (dstr), (sstr), (flags) & ~SV_NOSTEAL);  \
-    }                                                                    \
+#define sv_setsv_flags(dstr, sstr, flags)                                          \
+  STMT_START {                                                                     \
+    if (((flags) & SV_NOSTEAL) && (sstr) && (SvFLAGS((SV *)(sstr)) & SVs_TEMP)) {  \
```
Comment (atoomic) [inline parts/inc/Sv_set]: != NULL

**Comment by atoomic** [comment]
@pali at first glance need more love on Perl 5.6 as shown by Travis above

File: parts/inc/Sv_set [hunk 2]
```diff
@@ -160,6 +160,16 @@ newSVsv_nomg(sv)
 
 #endif
 
+void
+sv_setsv_compile_test(sv)
+        SV *sv
+        CODE:
+                sv_setsv(sv, NULL);
+                sv_setsv_flags(sv, NULL, 0);
```
Comment (pali) [inline parts/inc/Sv_set]: This should fix Perl 5.6 compilation which does not have `sv_setsv_flags` macro.
```suggestion
#ifdef sv_setsv_flags
                sv_setsv_flags(sv, NULL, 0);
#endif
```

File: parts/inc/Sv_set (hunk 1 above)
Comment (pali) [reply parts/inc/Sv_set]: That is implicit in `(sstr)`...

File: parts/inc/Sv_set (hunk 2 above)
Comment (atoomic) [reply parts/inc/Sv_set]: excellent giving it a try

**Comment by atoomic** [comment]
this is now good to be merged

**Pull request merged**